server/pulp/server/controllers/repository.py

```diff
@@ -205,26 +205,35 @@ def find_repo_content_units(
 
     """
 
-    qs = model.RepositoryContentUnit.objects(q_obj=repo_content_unit_q,
-                                             repo_id=repository.repo_id)
+    match = {"repo_id": repository.repo_id}
+    match.update(repo_content_unit_q.to_query(model.RepositoryContentUnit))
+    qs = model.RepositoryContentUnit.objects.aggregate(
+        {"$match": match},
+        {"$group": {"_id": "$unit_type_id", "name": {"$first": "$name"}}}
+    )
 
-    type_map = {}
+    type_list = []
     content_units = {}
 
     yield_count = 1
     skip_count = 0
 
     for repo_content_unit in qs:
-        id_set = type_map.setdefault(repo_content_unit.unit_type_id, set())
-        id_set.add(repo_content_unit.unit_id)
-        content_unit_set = content_units.setdefault(repo_content_unit.unit_type_id, dict())
-        content_unit_set[repo_content_unit.unit_id] = repo_content_unit
+        type_list.append(repo_content_unit["_id"])
 
-    for unit_type, unit_ids in type_map.iteritems():
+    for unit_type, unit_ids in type_list:
```

Comment:
@ggainey  Hi, I'm still in process of setuping pulp and pulp-smash. I assume the error could be caused by this line which I forgot to change properly.

Would it be possible to fix this and run pulp-smash on your side?

Reply:
Can do - will take a little while, my computer greeted me this morning by hard-locking and had to be rebooted, so it's going to take a few minutes to get everything working again :) Will comment here with results

Reply:
Ah, I see - yeah, get me a patch and I can re-run!

Reply:
Much better - we're down to `FAILED (failures=13, errors=14, skipped=16)`. Added test-output to the issue as `midnightercz-bson-too-large-fix-changed.txt.gz`

You def want to be firing up pulp-smash for yourself - see https://pulp-2-tests.readthedocs.io/en/latest/ for details.